watertap/unit_models/electrodialysis_1D.py

```diff
@@ -1787,18 +1863,24 @@ def _pressure_drop_calculation(self):
             units=pyunits.pascal,
             doc="pressure drop over an entire ED stack",
         )
-        self.friction_factor = Var(
-            initialize=10,
-            units=pyunits.dimensionless,
-            doc="friction factor of the channel fluid",
-        )
 
         if self.config.pressure_drop_method == PressureDropMethod.experimental:
             _log.warning(
                 "Do not forget to FIX the experimental pressure drop value in [Pa/m]!"
             )
         else:  # PressureDropMethod.Darcy_Weisbach is used
-            self._get_fluid_dimensionless_quantities()
+            if not (
+                self.config.has_Nernst_diffusion_layer
+                and self.config.limiting_current_density_method
+                == LimitingCurrentDensityMethod.Theoretical
+            ):
+                self._get_fluid_dimensionless_quantities()
```

Comment:
I'm surprised `black` doesn't force you to add a space here -- might be good for readability.